### PR TITLE
Rspec tests for messages and config options

### PR DESCRIPTION
**Pull request by marzann**
- Rspec tests for `Messages` have been added.
- dotenv gem was added to provide environment variables.
- attribute readers: `grouped_by` and `counted_by` were added to analyzeable.rb
- `attr_reader : messages` was added to messages.rb

### CODE REVIEW

File: spec/lib/facebook_data_analyzer/analyzeables/messages_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  before(:each) do
+    subject.analyze
+  end
+
+  describe '#analyze' do
+    it 'should return proper @messages' do
+      message = subject.messages.first
+
+      expect(message.character_count).to eq(304)
```
Comment (Lackoftactics) [inline spec/lib/facebook_data_analyzer/analyzeables/messages_spec.rb]: this test is failing with default example catalog. Also can't we setup Test Catalog to example by default for tests?

**Comment by marzann** [comment]
@Lackoftactics: Test Catalog path is updated. Local tests are passing. This PR should be ok now. 
Next, I am working on CI build, and will submit another PR soon.

**Comment by thnukid** [comment]
@marzann nicely done !

**Comment by marzann** [comment]
@thnukid: Thank you. I collaborated with @bodrovis on this, and want to thank him for his insights.
@Lackoftactics, @thnukid: are you able to run local tests successfully?

**Comment by thnukid** [comment]
I can run them successfully 👍 

```bash 
Randomized with seed 14977

FacebookDataAnalyzer::Messages
  #group
Analyzing Messages |Time: 00:00:00 | ======================================================================================================================================================================================================================================================================== | Time: 00:00:00
    should group messages by conversation correctly
  #count
Analyzing Messages |Time: 00:00:00 | ======================================================================================================================================================================================================================================================================== | Time: 00:00:00
    should count grouped messages correctly
  #analyze
Analyzing Messages |Time: 00:00:00 | ======================================================================================================================================================================================================================================================================== | Time: 00:00:00
    should return proper @messages

FacebookDataAnalyzer
  has a version number

FacebookDataAnalyzer::Contacts
  #analyze
    should return proper @contacts details for first contact

FacebookDataAnalyzer::Friends
  #analyze
    should return proper friends details
  #total
    should return correct friends count
  #count
    should count adding friends correctly

Finished in 0.09616 seconds (files took 0.5344 seconds to load)
8 examples, 0 failures

Randomized with seed 14977
```

**Comment by marzann** [comment]
I updated messages_spec, and the CI build is passing now. 👍

File: lib/facebook_data_analyzer.rb [hunk 2]
```diff
@@ -2,21 +2,21 @@
...
-  require 'facebook_data_analyzer/contact'
-  require 'facebook_data_analyzer/friend'
-  require 'facebook_data_analyzer/message'
-
-  require 'axlsx'
-  require 'parallel'
-  require 'json'
-  require 'workbook'
-  require 'set'
+require 'axlsx'
```
Comment (thnukid) [inline lib/facebook_data_analyzer.rb]: Out of curiosity, why is that moved outside of the `module FacebookDataAnalyzer` - is there some best practices?

Probably line 26 can be changed from `package = ::Axlsx::Package.new` to `package = Axlsx::Package.new`.
 Same with `b = ::Workbook::Book.open(xlsx)` on line 41

File: spec/lib/facebook_data_analyzer/analyzeables/friends_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    described_class.new(catalog: test_catalog)
+  end
+
+  before(:each) do
+    subject.analyze
+  end
+
+  describe '#analyze' do
+    it 'should return proper friends details' do
+      friend = subject.friends.first
```
Comment (thnukid) [inline spec/lib/facebook_data_analyzer/analyzeables/friends_spec.rb]: What I find a little bit nicer is, when you extract `friend` into a `let` statement.

```
  describe '#analyze' do
    let(:friend) { subject.friends.first }

    it 'should return proper friends details' do
      expect(friend.name).to eq('Isaac Perez')
      expect(friend.date_added.to_s).to eq("2017-11-05T00:00:00+00:00")
```

File: spec/lib/facebook_data_analyzer/analyzeables/friends_spec.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      expect(friend.name).to eq('Isaac Perez')
+      expect(friend.date_added.to_s).to eq("2017-11-05T00:00:00+00:00")
+    end
+  end
+
+  describe '#total' do
+    it 'should return correct friends count' do
+      all_friends = subject.friends
```
Comment (thnukid) [inline spec/lib/facebook_data_analyzer/analyzeables/friends_spec.rb]:
```
let(:all_friends) { subject.friends }

File: lib/facebook_data_analyzer.rb (hunk 2 above)
Comment (bodrovis) [reply lib/facebook_data_analyzer.rb]: Yeah, it is quite common  to do so. For instance: https://github.com/sferik/twitter/blob/master/lib/twitter.rb (I  often use this gem as a reference because it was  written by a team of very experienced guys).

Comment (bodrovis) [reply lib/facebook_data_analyzer.rb]: To elaborate, there is no reason to `require` inside a module, because the required constants (including classes/modules which are constants too) will be available globally anyways.

File: spec/lib/facebook_data_analyzer/analyzeables/friends_spec.rb (hunk 3 above)
Comment (marzann) [reply spec/lib/facebook_data_analyzer/analyzeables/friends_spec.rb]: @thnukid: 👍 I will extract friend, contact and message in spec files into lets.

File: lib/facebook_data_analyzer.rb (hunk 2 above)
Comment (marzann) [reply lib/facebook_data_analyzer.rb]: @thnukid: I will update lines 26 and 41, as you suggested.  
I tracked line 26 to: 7f1b8d09 (Eric Bartholemy 2018-05-05 00:04:50 +0200 26)     package = ::Axlsx::Package.new, so I can safely assume the contributor will not mind. :)
I cannot easily track the origin of current line 41, due to file renames and frequent commits, but will update it as well, for consistency.

Comment (thnukid) [reply lib/facebook_data_analyzer.rb]: Good that I have not left any tracks on line 41 :D

Comment (marzann) [reply lib/facebook_data_analyzer.rb]: Lol! No worries! This is a good update.

**Comment by thnukid** [comment]
This LGTM and is mergeable.

Their are other PRs #31 #32  that build on this PR.
